src/msg/async/rdma/RDMAStack.cc

```diff
@@ -333,6 +336,22 @@ RDMAConnectedSocketImpl* RDMADispatcher::get_conn_lockless(uint32_t qp)
   return it->second.second;
 }
 
+Infiniband::QueuePair* RDMADispatcher::get_qp(uint32_t qp)
+{
+  Mutex::Locker l(lock);
+  // Try to find the QP in qp_conns firstly.
+  auto it = qp_conns.find(qp);
+  if (it == qp_conns.end()) {
+    // Try again in dead_queue_pairs.
+    for(auto dead_qp = dead_queue_pairs.begin(); dead_qp != dead_queue_pairs.end(); dead_qp++) {
```

Comment:
might want to use range-based loop or `std::find_if()` instead.

Reply:
The same thing that may need your review:

  if (it != qp_conns.end())
    return it->second.first;

  for (auto &i : dead_queue_pairs)
    if (i->get_local_qp_number() == qp)
      return i;

  return nullptr;